<launch>
	<!-- global rosparam -->
	<param name="use_sim_time" value="true"/>

	<!-- rosbag -->
	<node pkg="rosbag" type="play" name="player" args="--clock /home/amsl/ozaki/bagfiles/sq2/indoor_dkan/2018-05-08-19-45-05.bag"/>

	<!-- rviz -->
	<node pkg="rviz" type="rviz" name="rviz" args="-d $(find scan_matching)/rviz_config/scan_matching_sqlidar.rviz"/>

	<!-- semi sensor driver -->
	<node pkg="sq_lidar" name="sq_lidar" type="sq_lidar">
		<param name="fixed_frame" value="/base_link"/>
		<remap from="~/scan" to="/sq_lidar/scan"/>
		<remap from="~/cloud" to="/cloud"/>
	</node>
	<node pkg="sq_lidar_pc_store" name="sq_lidar_pc_store" type="sq_lidar_pc_store" output="screen">
		<param name="num_scans" type="int" value="2"/>
	</node>
	<node pkg="gyrodometry" name="initial_alignment" type="initial_alignment" output="screen">
		<param name="initial_orientation_is_0001" type="bool" value="true"/>
	</node>

	<!-- main -->
	<node pkg="scan_matching" name="scan_matching_ekf" type="scan_matching_ekf" output="screen">
		<param name="child_frame_name" type="string" value="/ekf/odom"/>
		<param name="sigma_imu" type="double" value="1e-5"/>
		<param name="sigma_odom" type="double" value="1e-5"/>
		<param name="sigma_ndt" type="double" value="1e+1"/>
	</node>
	<node pkg="scan_matching" name="ndt_scan_matching" type="ndt_scan_matching" output="screen">
		<remap from="/velodyne_points" to="/sq_lidar/stored"/>
		<param name="pc_range" type="double" value="20.0"/>
		<param name="leafsize_source" type="double" value="0.15"/>
		<param name="leafsize_target" type="double" value="0.2"/>
		<param name="trans_epsilon" type="double" value="1e-3"/>
		<param name="stepsize" type="double" value="0.1"/>
		<param name="resolution" type="double" value="0.5"/>
		<param name="max_iterations" type="int" value="30"/>
	</node>

	<!-- comparisons -->
	<node pkg="gyrodometry" name="gyrodometry3d" type="gyrodometry3d"/>

	<!-- for visualize trajectory -->
	<node pkg="msg_conversion" type="posestamped_to_odometry" name="posestamped_to_odometry">
		<remap from="/pose" to="/ekf/pose"/>
		<remap from="/odom/from_posestamped" to="/ekf/odom"/>
		<param name="child_frame_id" type="string" value="/ekf/odom"/>
	</node>
	<node pkg="msg_conversion" type="pc_rename_frameid" name="pc_rename_frameid">
		<remap from="/cloud" to="/sq_lidar/stored"/>
		<param name="frame_name" type="string" value="/ekf/odom"/>
	</node>
</launch>
